pkg/virt-controller/watch/migration/migration.go

```diff
@@ -1060,11 +1123,50 @@ func (c *Controller) handleTargetPodCreation(key string, migration *virtv1.Virtu
 				return nil
 			}
 		}
+		err = c.handleBackendStorage(migration, vmi)
```

Comment:
It seems there's nothing preventing to call this twice while the controller is waiting for the informer to pick up the PVC creation..? This could lead to creating the pvc twice.

Example: you go through `handleTargetPodCreation`, `handleBackendStorage` gets called and the first pvc get created, then you immediately call `createTargetPod` which returns early because the pvc informer doesn't have the new pvc yet (line 706). Eventually you'll go again through `handleTargetPodCreation` which is going to create the second pvc.

Please double check whether I'm wrong.

To fix this I think you should check the pvc expectations on line 1060 as well, inside `handleTargetPodCreation`. You also need to make sure that `handleBackendStorage` doesn't create the pvc again if it finds one in the informer. This way you're sure that whenever you call `handleBackendStorage` either the pvc has been created and the informer is updated or the pvc has not been created yet.

Reply:
I think it's ok for `handleBackendStorage` to be called multiple times, since it won't create a new PVC if `TargetPersistentStatePVCName` is set, right?
I am however missing a call to `SatisfiedExpectations` in `execute()` on the needsSync line, will add. (that probably also addresses your concern?).

Re-reading the code for `handleBackendStorage` though, I think we're not handling RWX storage correctly, as we're setting an expectation that will never be satisfied! The code probably works anyway because of the missing call to `SatisfiedExpectations`. i.e. this bug and the previous cancel each other!

Will fix, thanks!